src/settled-cash/CrossMarginCashEngine.sol

```diff
@@ -85,6 +89,16 @@ contract CrossMarginCashEngine is
     ///      assetId => assetId masks
     mapping(uint256 => uint256) private collateralizable;
 
+    /// @dev nonce for signed messages to prevent replay attacks
+    ///      address => nonce
+    mapping(address => uint256) private nonces;
+
+    /// @dev initial chain id used in domain separator calculations
+    uint256 private initialChainId;
+
+    /// @dev initial cached domain separator
+    bytes32 private initialDomainSeparator;
```

Comment:
we need a one time function to set this after upgrade. Callable only if this hasn't been set previously